crates/mapdiffbot2/src/job_processor.rs

```diff
@@ -361,34 +261,27 @@ pub fn do_job(job: &Job) -> Result<CheckOutputs> {
         .to_str()
         .ok_or_else(|| anyhow::anyhow!("Failed to create absolute path to image directory",))?;
 
-    git_checkout(
-        &job.base
-            .repo
-            .default_branch
-            .clone()
-            .unwrap_or_else(|| "master".to_owned()),
-    )
-    .context("Checking out to default branch")?; // If this fails, good luck
-
     let filter_on_status = |status: FileDiffStatus| {
         job.files
             .iter()
             .filter(|f| f.status == status)
             .collect::<Vec<&FileDiff>>()
     };
 
+    let diffs = git2::Diff::from_buffer(job.patch.as_ref().unwrap().as_bytes())
+        .map_err(|_| anyhow::anyhow!("Failed to parse patch, probably corrupted"))?;
```

Comment:
Does this work here?
```suggestion
        .context("Failed to parse patch, probably corrupted")?;
```